### DISCUSSION TITLE
Allow hook into Popover.Button onClick

### DISCUSSION THREAD

**Discussion by jondeibel** [tailwindlabs/headlessui#548] [Ideas]
Right now in @headlessui/react's Popover.Button you gobble up any reference to the onClick on the Popover.Button. I have a case where I am basically building a popconfirm on a destructive action in the UI with a popover. If the user hits YES in that popover then it should trigger some network call and close the popover. Is there a reason the passed onClick is overrode rather than called along with the internal one?

**Comment by sebastienbarre** [comment]
I second this, I would like to be able to trigger my own `onClick` before Popover.Button's own `handleClick` logic, so I can override the Popper JS positioning logic dynamically before the popper itself is displayed (it can't be done with `usePopper`'s static modifiers). 
Thanks

**Comment by dcastil** [comment]
I have the same issue as well. It would be nice if Headless UI would "wrap" callbacks it overrides. For the meantime I found this workaround:

```tsx
<Popover.Button as={Fragment}>
    <button onClick={myOwnHandleClick}>
        …
    </button>
</Popover.Button>
```

**Reply by mfyz** [reply]
@dcastil I was able to use the "close" prop and call that wherever I want inside the panel.
See: https://headlessui.dev/react/popover#closing-popovers-manually
